scapy/utils.py

```diff
@@ -1328,7 +1329,7 @@ class RawPcapWriter:
     """A stream PCAP writer with more control than wrpcap()"""
 
     def __init__(self, filename, linktype=None, gz=False, endianness="",
-                 append=False, sync=False, nano=False):
+                 append=False, sync=False, nano=False, snaplen=MTU):
```

Comment:
Question here: what would snaplen=0 do? if it is equivalent to snaplen=infinite, I'd prefer if we used it as default value.

Reply:
I don't think it matters. The per-file snaplen is kind of informative in pcap, as every packet has its own snaplen. I was trying not to change anything, in case it matters (e.g. tests that care). I suggest to change it to 262144 per [this](https://github.com/the-tcpdump-group/tcpdump/blob/master/netdissect.h#L335), but in a follow-up patch. 

```
/*
 * Maximum snapshot length.  This should be enough to capture the full
 * packet on most network interfaces.
 *
 *
 * Somewhat arbitrary, but chosen to be:
 *
 *    1) big enough for maximum-size Linux loopback packets (65549)
 *       and some USB packets captured with USBPcap:
 *
 *           https://desowin.org/usbpcap/
 *
 *       (> 131072, < 262144)
 *
 * and
 *
 *    2) small enough not to cause attempts to allocate huge amounts of
 *       memory; some applications might use the snapshot length in a
 *       savefile header to control the size of the buffer they allocate,
 *       so a size of, say, 2^31-1 might not work well.
 *
 * XXX - does it need to be bigger still?  Note that, for versions of
 * libpcap with pcap_create()/pcap_activate(), if no -s flag is specified
 * or -s 0 is specified, we won't set the snapshot length at all, and will
 * let libpcap choose a snapshot length; newer versions may choose a bigger
 * value than 262144 for D-Bus, for example.
 */
#define MAXIMUM_SNAPLEN 262144
```